pkg/velero/backup.go

```diff
@@ -0,0 +1,353 @@
+		crd, builder.Definition.Name, builder.Definition.Namespace)
+
+	if crd == "" {
+		glog.V(100).Infof("Backup excluded cluster-scoped resource is empty")
+
+		builder.errorMsg = "backup excluded cluster-scoped resource cannot be an empty string"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.ExcludedClusterScopedResources =
+		append(builder.Definition.Spec.ExcludedClusterScopedResources, crd)
+
+	return builder
+}
+
+// WithExcludedNamespaceScopedResources adds the specified namespace-scoped crd for exclusion when performing a backup.
+func (builder *BackupBuilder) WithExcludedNamespaceScopedResources(crd string) *BackupBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Adding customresource %s to backup %s in namespace %s excluded namespace-scoped resources",
+		crd, builder.Definition.Name, builder.Definition.Namespace)
+
+	if crd == "" {
+		glog.V(100).Infof("Backup excluded namespace-scoped resource is empty")
+
+		builder.errorMsg = "backup excluded namespace-scoped resource cannot be an empty string"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.ExcludedNamespaceScopedResources =
+		append(builder.Definition.Spec.ExcludedNamespaceScopedResources, crd)
+
+	return builder
+}
+
+// Exists checks whether the given backup exists.
+func (builder *BackupBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if backup %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.apiClient.VeleroV1().Backups(builder.Definition.Namespace).Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create makes a backup according to the backup definition and stores the created object in the backup builder.
+func (builder *BackupBuilder) Create() (*BackupBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating backup %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.VeleroV1().Backups(builder.Definition.Namespace).Create(
+			context.TODO(), builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Update renovates the existing backup object with the backup definition in builder.
+func (builder *BackupBuilder) Update() (*BackupBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating backup %s in namespace %s", builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.apiClient.VeleroV1().Backups(builder.Definition.Namespace).Update(
```

Comment:
```suggestion
	builder.Object, err := builder.apiClient.VeleroV1().Backups(builder.Definition.Namespace).Update(
```

Reply:
I get this error with that: `expected identifier on left side of :=` since builder.Object already exists

Reply:
Sounds good!